### PR TITLE
Create Virtual Serial Simulation

### PR DESCRIPTION
**Pull request by goobta**
Create a realtime solo environment and be able to communicate with it over Serial. Note that this requires that https://github.com/WPI-MMR/gym_solo/pull/44 is merged in.

To get it working, there are two convenience scripts to get ya going (in the `scripts` folder). Run the `solo8...py` script first, which should open the PyBullet GUI and tell you which serial port to listen on. Then run `interactive.py` and attach to that aforementioned serial port. 

You can then type in the following commands:
- `RESET`
- `OBS`

`ACTION` is also implemented, but you need to pass an encoded JSON string, which is too much effort imo.

### CODE REVIEW

**Comment by goobta** [comment]
if you wanna make the robot dance, paste the follow string into `interactive.py`!

```
ACTION{"\\u03b8x": -0.282183378500551, "\\u03b8y": 0.4477730498313323, "\\u03b8z": -0.7515055486502317, "vx": -0.0021985155092427927, "vy": -0.0016372307421560893, "vz": 0.0010710493053103162, "wx": 0.00677695439096879, "wy": -0.011411640559245578, "wz": 0.002068529012777597, "FL_HFE": 3.9571158441228693, "FL_KFE": -3.705995755504008, "FL_ANKLE": 0.0, "FR_HFE": -6.240330487904615, "FR_KFE": 4.285736406870781, "FR_ANKLE": 0.0, "HL_HFE": -0.32403276755326144, "HL_KFE": -5.4244141069928276, "HL_ANKLE": 0.0, "HR_HFE": -4.428453979549971, "HR_KFE": -2.745596111102513, "HR_ANKLE": 0.0}
```

File: ros2sim/parsers/json_parser.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    self._COMMAND_MAPPING = {
+      parsers.special.RESET: self.reset,
+      parsers.special.OBS_REQUEST: self.get_obs,
+      parsers.special.ACTION: self.action,
+    }
+
+  def parse(self, input_str: str) -> str:
+    resp = None
+    for cmd, func in self._COMMAND_MAPPING.items():
+      if input_str.startswith(cmd.value):
```
Comment (mahajanrevant) [inline ros2sim/parsers/json_parser.py:23]: I am assuming in each parser call, this if statement will be true only once?

File: ros2sim/parsers/test_base_parser.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import unittest
+
+from ros2sim.parsers import Parser
+
+
+class DummyParser(Parser):
+  def parse(input_str):
+    pass
+
```
Comment (mahajanrevant) [inline ros2sim/parsers/test_base_parser.py:9]: Nit: One extra line

**Review by mahajanrevant — COMMENTED** [review]
Small comments

File: ros2sim/parsers/json_parser.py (hunk 1 above)
Comment (goobta) [reply ros2sim/parsers/json_parser.py:23]: I'm assuming each call corresponds to a "command". Therefore, if there are two commands within the same input string, it'll only parse the first one.

File: ros2sim/parsers/test_base_parser.py (hunk 2 above)
Comment (goobta) [reply ros2sim/parsers/test_base_parser.py:9]: done

**Review by goobta — COMMENTED** [review]
Next commit should have all requested changes.

**Review by mahajanrevant — APPROVED** [review]
LGTM

**Comment by mahajanrevant** [comment]
This is a cleaned up string to execute as a test: ACTION{ "FL_HFE": 3.9571158441228693, "FL_KFE": -3.705995755504008, "FL_ANKLE": 0.0, "FR_HFE": -6.240330487904615, "FR_KFE": 4.285736406870781, "FR_ANKLE": 0.0, "HL_HFE": -0.32403276755326144, "HL_KFE": -5.4244141069928276, "HL_ANKLE": 0.0, "HR_HFE": -4.428453979549971, "HR_KFE": -2.745596111102513, "HR_ANKLE": 0.0}

**Pull request merged**